### PR TITLE
Ord for entry kind, small api improvements

### PR DESCRIPTION
**Pull request by anti-social**
There were some api changes:
- `Entry::get_path` was renamed to `Entry::path`
- `Entry::kind` now returns `EntryKind<&Path>` (don't know why it returned `EntryKind<PathBuf>`)

But there was not release with the above api so I think we can change it.

### CODE REVIEW

File: src/v1/parser.rs [hunk 1]
```diff
@@ -124,14 +124,99 @@ quick_error! {
...
+    pub fn as_ref(&self) -> EntryKind<&Path> {
+        use self::EntryKind::*;
+        match *self {
+            Dir(ref p) => Dir(p.as_ref()),
+            File(ref p) => File(p.as_ref()),
+        }
+    }
+
+    /// Clones path and returns EntryKind<PathBuf>
+    pub fn cloned(&self) -> EntryKind<PathBuf> {
```
Comment (tailhook) [inline src/v1/parser.rs]: maybe `to_owned`?

File: src/v1/parser.rs [hunk 2]
```diff
@@ -124,14 +124,99 @@ quick_error! {
...
+}
+
+impl<P> Ord for EntryKind<P>
+    where P: AsRef<Path> + PartialEq + Eq
+{
+    fn cmp(&self, other: &EntryKind<P>) -> Ordering {
+        use std::cmp::Ordering::*;
+        use self::EntryKind::*;
+
+        match *self {
```
Comment (tailhook) [inline src/v1/parser.rs]: Built-in comparison of paths is by component. So I believe ord should look like:
```
self.path().cmp(other.path()).and_then(|| match (self, other) {
  (&Dir(_), &Dir(_)) => Equal,
  (&Path(_), &Path(_)) => Equal,
  (&Dir(_), &Path(_)) => Less,
  (&Path(_), &Dir(_)) => Greater,
})
```

File: src/v1/parser.rs (hunk 1 above)
Comment (anti-social) [inline src/v1/parser.rs]: `to_owned` is the method of the `ToOwned` trait. So I think we shouldn't use that name outside of `ToOwned` implementation.

File: src/v1/parser.rs (hunk 2 above)
Comment (anti-social) [inline src/v1/parser.rs]: That won't work. For the signature file format `File("/1") < Dir("/1")`, but if you first compare paths it will be equal.

File: src/v1/parser.rs (hunk 1 above)
Comment (anti-social) [inline src/v1/parser.rs]: `Option` has similar `as_ref` and `cloned` methods.

File: src/v1/parser.rs (hunk 2 above)
Comment (tailhook) [inline src/v1/parser.rs]: This is the whole point of `Ordering::and_then`

Comment (tailhook) [inline src/v1/parser.rs]: Oops. `Ordering::then`

Comment (anti-social) [inline src/v1/parser.rs]: I've tested your code. It doesn't work:
```
assertion failed: `(left == right)` (left: `Less`, right: `Greater`): e1: File("/1"), e2: Dir("/1")
```

If we change match to `(&Dir(_), &Path(_)) => Greater` and `(&Path(_), &Dir(_)) => Less` another will fail:
```
assertion failed: `(left == right)` (left: `Less`, right: `Greater`): e1: File("/a"), e2: Dir("/1")
```

So we cannot compare paths directly.

File: src/v1/parser.rs (hunk 1 above)
Comment (tailhook) [inline src/v1/parser.rs]: Okay

File: src/v1/parser.rs (hunk 2 above)
Comment (tailhook) [inline src/v1/parser.rs]: Yes, so basically I have misunderstood comparison rules. I'll take another look at it later, maybe I can find a better way to express this sort order. Or at least some comments might be added.

File: src/v1/parser.rs [hunk 3]
```diff
@@ -123,15 +123,105 @@ quick_error! {
...
+                            Greater | Equal => Greater,
+                        }
+                    },
+                    File(ref other_path) => {
+                        let other_parent = other_path.as_ref().parent()
+                            .unwrap_or(Path::new("/"));
+                        match parent.cmp(other_parent) {
+                            Less => Less,
+                            Greater => Greater,
+                            Equal => path.as_ref().cmp(other_path.as_ref()),
```
Comment (tailhook) [inline src/v1/parser.rs]: Will you fix this to filename comparison?

Comment (anti-social) [inline src/v1/parser.rs]: `Path.file_name()` returns `Option` and it breaks corner case test when comparing `File("")` and `File("/")`.

Comment (tailhook) [inline src/v1/parser.rs]: Well, it's impossible to represent empty directory (only root), because that would be an empty line. And it's impossible to represent File that has an empty filename, because we would parse size as a filename in this case.

So I believe we can assert on it.
